### DISCUSSION TITLE
How to generate tests for all enums defined in a specification

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#502] [Q&A] [answered]
Hi there, 

Is it possible in portman to generate tests which include all possible values defined in an enum. For example, I want to generate 5 tests which use each value in my salutation enum. 

![image](https://github.com/apideck-libraries/portman/assets/5680263/db9b0896-ba70-4db3-9064-787aeff83468)

Thanks in advance,
Paul.

**Comment by thim81** [comment, marked as answer]
Hi @PaulCluver 

The schema validation will use the ENUM values to validate if the response contains any of the allowed values.

Next to this, you could manually create a contentTests > responseBodyTests using the `oneOf` property, to manually validate the content.
https://github.com/apideck-libraries/portman#portman---contenttests-properties

**Reply by PaulCluver** [reply]
Thanks for the advice. Adding a responseBodyTest with the oneOf property works fine.
